class-components-todo/src/App.js

```diff
@@ -0,0 +1,168 @@
+import React, { Component } from "react";
+import './App.css';
+import uuid from 'react-uuid';
+import { FaChevronDown } from "react-icons/fa";
+import  Button  from "@mui/material/Button";
+
+class App extends Component{
+  constructor(props) {
+    super(props);
+
+    // getInitialState
+    this.state = {
+      all: [],
+  newToDo:"",
+  mode:"all",
+  editToDo:"",
+  editToDoId:""
+    };
+  }
+
+  handleChange=(event)=>{
+    this.setState({newToDo:event.target.value});
+  };
+
+  handleKeypress = e => {
+    //it triggers by pressing the enter key
+  if (e.key==='Enter') {
+    this.newTodo();
+    this.setState({newToDo:""});
+  }
+};
+
+  newTodo=()=>{
+    this.setState({
+      all:[
+        ...this.state.all,
+        {id:uuid(),action:this.state.newToDo,done:false,isEdit:false}
+      ]});
+  };
+
+
+  toggleDone = (todo) =>
+  this.setState({
+    all: this.state.all.map((item) =>
+      item.action === todo.action ? { ...item, done: !item.done } : item
+    ),
+  });
+
+  delete=(todo)=>{
+    this.setState({ all: this.state.all.filter((item) => this.state.all.indexOf(item) !== this.state.all.indexOf(todo)) });
+  }
+
+  handleEdit= e =>{
+    this.setState({editToDo : e.target.value,editToDoId:e.target.id});
+  };
+
+  editTodo=()=>{
+    this.setState({
+      all: this.state.all.map((item) =>
+        item.id === this.state.editToDoId ? { ...item, action:this.state.editToDo ,isEdit:false} : item
+      )
+    });
+  }
+
+
+  handleKeyEdit = e => {
+    //it triggers by pressing the enter key
+  if (e.key==='Enter') {
+    this.editTodo();
+    this.setState({editToDo:"",editToDoId:""});
+  }
+};
+
+
+
+listToMap=()=> {
+  if(this.state.mode==="active")
+  {
+    return this.state.all.filter(item=>item.done===false);
+  }
+  else if(this.state.mode==="completed"){
+    return this.state.all.filter(item=>item.done===true);
+  }
+  else{
+    return this.state.all;
+  }
+  }
+
+
+edit = (todo) => {
+  this.setState({
+    all: this.state.all.map((item) =>
+      item.action === todo.action ? { ...item, isEdit: !item.isEdit } : item
+    ),
+    editToDo: todo.action
+  });
+}
+  
+
+  btnClick = () =>{
+    if(this.countCompleted()<this.count()){
+  this.setState({ all : this.state.all.map((item) =>
+  item = { ...item, done: true })})
+  }
+  else{
+    this.setState({ all : this.state.all.map((item) =>
+      item = { ...item, done: false })})
+  }
+}
+
+active=()=>{
```

Comment:
no need to create three separate functions(active, completed, showAll), instead you can create a function setMode and pass the mode string("active"/"completed"/"all") as arguments